.github/workflows/lint-python.yml

```diff
@@ -0,0 +1,51 @@
+name: Lint Python
```

Comment:
Maybe we can use an pre-built action - https://github.com/wearerequired/lint-action

?

Reply:
I'm not familiar with that action; what does it help with? My original idea was to try and align our repo setup as much as possible with polars upstream